server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -117,6 +117,7 @@ public abstract class Engine implements Closeable {
     public static final String CAN_MATCH_SEARCH_SOURCE = "can_match";
     protected static final String DOC_STATS_SOURCE = "doc_stats";
     public static final long UNKNOWN_PRIMARY_TERM = -1L;
+    protected static final ThreadLocal<Boolean> IS_FLUSH_BY_REFRESH = ThreadLocal.withInitial(() -> false);
```

Comment:
Looks like it is used only in a specific engine implementation, so maybe it can be defined there?

Reply:
Yep. Moved this to IndexEngine.